doc/development_guide.md

```diff
@@ -12,134 +13,336 @@ a CLA and decorate the PR appropriately (e.g., label, comment). Simply follow th
+If `<workspaceRoot>/.vscode` does not exists, change one of the user settings to create it. One of the easiest way is simply changing the current version of python to any version.
+
+![image](https://user-images.githubusercontent.com/19577035/63064397-b9f96c80-beb5-11e9-8514-b0e58e83f3a5.png)
+
+And you will see `<workspaceRoot>/.vscode` and `<workspaceRoot>/.vscode/settings.json` file are created.
+
+![image](https://user-images.githubusercontent.com/19577035/63064590-bca89180-beb6-11e9-8800-1bcde427cef0.png)
+
+Update the `<workspaceRoot>/.vscode/settings.json` as shown below:
+##### Windows
+```
+{
+    "python.pythonPath": "${workspaceRoot}/env/Scripts/python.exe",
+    "python.linting.pylintEnabled": true,
+    "python.envFile": "${workspaceRoot}/environments.env",
+    "python.testing.nosetestsEnabled": true,
+    "python.testing.nosetestArgs": [
+        "tests",
+        "mssqlcli/jsonrpc/tests",
+        "mssqlcli/jsonrpc/contracts/tests"
+    ],
+    "python.testing.pytestEnabled": false,
+    "python.testing.unittestEnabled": false
+}
+```
+##### MacOS / Linux
+```
+{
+    "python.pythonPath": "${workspaceRoot}/env/bin/python",
+    "python.linting.pylintEnabled": true,
+    "python.envFile": "${workspaceRoot}/environments.env",
+    "python.testing.nosetestsEnabled": true,
+    "python.testing.nosetestArgs": [
+        "tests",
+        "mssqlcli/jsonrpc/tests",
+        "mssqlcli/jsonrpc/contracts/tests"
+    ],
+    "python.testing.pytestEnabled": false,
+    "python.testing.unittestEnabled": false
+}
+```
+
+Press `Yes` if any notification comes up for missing dependencies. Please make sure to complete [Build mssql-cli](#Build_mssql_cli) step before moving forward.
+
+![image](https://user-images.githubusercontent.com/19577035/63066150-9b976f00-bebd-11e9-8e31-a786e2eb61f5.png)
+
+
+
+### 3. Create environments.env file
+Create a file named `<workspaceRoot>/environments.env` with contents:
+```
+MSSQL_CLI_SERVER=<hostname>
+MSSQL_CLI_DATABASE=<database>
+MSSQL_CLI_USER=<username>
+MSSQL_CLI_PASSWORD=<password>
+```
+
+![image](https://user-images.githubusercontent.com/19577035/63066435-e82f7a00-bebe-11e9-9ef0-4b1f78a8e38b.png)
+
+
+## <a name="How_To_Run_mssql_cli_With_Debugger"></a> 3. How To Run mssql_cli With Debugger
+### 1. Configure launch.json
+Go to debug window by clicking debug icon on the left.
+If you have no configuration, click the gear icon with red circle badge to create `<workspaceRoot>/.vscode/launch.json` file.
+
+![image](https://user-images.githubusercontent.com/19577035/63065387-b61c1900-beba-11e9-8cb4-2994602ce8ea.png)
+
+Select `Python` for environment.
+
+![image](https://user-images.githubusercontent.com/19577035/63065581-70138500-bebb-11e9-9422-2f24f2bca467.png)
+
+Select `Python File` for debug configuration.
+
+![image](https://user-images.githubusercontent.com/19577035/63065617-99ccac00-bebb-11e9-8f2a-ba1f745551d3.png)
+
+Update the `<workspaceRoot>/.vscode/launch.json` as shown below:

```

Comment:
Since we're checking this in, we can remove this step.

Reply:
fixed.